experiment_pages/experiment/experiment_menu_ui.py

```diff
@@ -31,7 +31,10 @@ def __init__(self, parent: CTk, name: str, prev_page: ChangeableFrame = None, co
         self.data_page = DataCollectionUI(parent, self, name)
         self.analysis_page = DataAnalysisUI(parent, self)
         self.cage_page = CageConfigurationUI(name, parent, self)
-        self.rfid_page = MapRFIDPage(name, parent, self, controller)
+        if controller is None:
```

Comment:
This change works on my local machine, allowing for the code to work without a controller attached. This was giving me an issue of too many arguments passed to the function otherwise.

Reply:
I would leave a comment here explaining why this code is necessary in the scenario of having no serial port controller.